### ISSUE TITLE
Possible empty list in qterminal/src/termwidget.cpp::TermWidgetImpl().

### ISSUE THREAD

**Issue by textbrowser** [lxqt/qterminal#841]
```
QString TerminalConfig::getShell()
{
    if (!m_shell.isEmpty())
        return m_shell;
    if (!Properties::Instance()->shell.isEmpty())
        return Properties::Instance()->shell;
    QByteArray envShell = qgetenv("SHELL");
    if (envShell.constData() != nullptr)
    {
        QString shellString = QString::fromLocal8Bit(envShell);
        if (!shellString.isEmpty())
            return shellString;
    }
    return QString();
}
```
may return a non-empty string which contains only spaces. For example $SHELL is configured incorrectly ("     ").
Then,
```
 QString shell = cfg.getShell();
    if (!shell.isEmpty())
    {
        //qDebug() << "Shell program:" << shell;
        QStringList parts = shell.split(QRegExp(QStringLiteral("\\s+")), Qt::SkipEmptyParts);
        //qDebug() << parts;
        setShellProgram(parts.at(0));
        parts.removeAt(0);
        if (parts.count())
            setArgs(parts);
    }
```
will fail at parts.at(0) because parts will be empty.

##### Steps to Reproduce (for bugs)

1. 
2. 
3. 
4. 

##### System Information

* Distribution & Version: 
* Kernel: 
* Qt Version: 
* qtermwidget Version: 
* lxqt-build-tools Version: 
* Package version:

**Comment by yan12125** [comment]
Nice catch! Do you want to create a pull request? Maybe trimming spaces can avoid such issues.

**Comment by textbrowser** [comment]
Will try today.

**Comment by textbrowser** [comment]
Please see https://github.com/lxqt/qterminal/pull/842.

**Comment by yan12125** [comment]
Fixed with #842, thanks!
